python/ray/tune/ray_trial_executor.py

```diff
@@ -212,16 +287,31 @@ def _setup_remote_runner(self, trial, reuse_allowed):
             logger.debug("Cannot reuse cached runner {} for new trial".format(
                 self._cached_actor))
             with self._change_working_directory(trial):
-                self._trial_cleanup.add(trial, actor=self._cached_actor)
+                pg = self._pg_manager.get_trial_pg(trial)
+
+                self._trial_cleanup.add(
+                    trial, actor=self._cached_actor, placement_group=pg)
+                self._pg_manager.clean_trial_placement_group(trial)
             self._cached_actor = None
 
         _actor_cls = _class_cache.get(trial.get_trainable_cls())
-        full_actor_class = _actor_cls.options(
-            num_cpus=trial.resources.cpu,
-            num_gpus=trial.resources.gpu,
-            memory=trial.resources.memory or None,
-            object_store_memory=trial.resources.object_store_memory or None,
-            resources=trial.resources.custom_resources)
+        if trial.uses_placement_groups:
+            if not self._pg_manager.is_ready(trial):
+                if not self._pg_manager.is_staging(trial):
+                    self._pg_manager.stage_trial(trial)
+
+            full_actor_class = self._pg_manager.get_full_actor_cls(
+                trial, _actor_cls)
+            if not full_actor_class:
```

Comment:
will this always return if the pg is ready?

if so, maybe we can avoid special-casing out the `if not full_actor_class` and always assume it returns True?

Reply:
So like this 
```
else:
   full_actor_class = self. ...
```
?

Reply:
yeah i think so?